src/Controls/src/Core/Platform/Windows/CollectionView/ItemContentControl.cs

```diff
@@ -190,10 +190,36 @@ internal void Realize()
 				_visualElement.BindingContext = dataContext;
 			}
 
-			Content = _renderer.ToPlatform();
+			if (_renderer.VirtualView is ICrossPlatformLayout)
+			{
+				Content = _renderer.ToPlatform();
+			}
+			else
+			{
+				Content = new LayoutBacking(_renderer.VirtualView);
```

Comment:
Fundamentally I think we need the measureoverride and arrangeoverride of the content control to just be able to flow naturally. There's a lot of behavior inside the base class of ContentControl that has to execute so we just need the content to be set to a control type that's able proxy the measure/override up to IView

If you look at the measureoverride/measurearrange now we can basically just remove all the code.

I found that if you don't call base.measureoverride on ContentControl then it doesn't add the Content to the ContentPresenter and the measuring gets weird. That's why the current implementation first calls base and then does a second measure pass by calling frameworkElement.Measure. If we just set the content to a control that knows how to reach up to IView then we only need one pass of measure/arrange

Later down the road once we go to ItemRepeater I wonder if ContentControl is the right base class to use here.

Reply:
Doesn't this add a new UIView for every cell ?

Reply:
Only if the view being added doesn't have a platform view that already calls up into the xplat. 

If the DataTemplate for a CV is something like a `Button`

I don't know a way to  make that work without adding a view that's MAUI layout aware.